ops/model.py

```diff
@@ -230,6 +241,96 @@ def _get_unique(self, relation_name, relation_id=None):
             raise TooManyRelatedAppsError(relation_name, num_related, 1)
 
 
+class BindingMapping:
+    """A map of binding names to lists of Binding instances."""
+
+    def __init__(self, backend):
+        self._backend = backend
+        self._data = {}
+
+    def get(self, binding_name):
+        if isinstance(binding_name, Relation):
+            key = binding_name
+            name = binding_name.name
+            relation_id = binding_name.id
+        elif isinstance(binding_name, str):
+            name = key = binding_name
```

Comment:
This type needs a bit of love as a whole. The idea is pretty simple, yet it manages to mix things up somewhat.

The map should be more consistent, and the variable names should be more indicative of what they hold. binding_name.name? binding_name.id? name = binding_name?  There's no other name being handled by this method other than binding names, so why do we have two binding names to hold the same name? Etc.

Reply:
Addressed in https://github.com/canonical/operator/pull/109/commits/10eeb01cf71dceaa69132d457a0f34eef09a6c2a